### PR TITLE
Add function to search Dash.app

### PR DESCRIPTION
**Pull request by tonylotts**
Plugin for searching Dash.app

### CODE REVIEW

**Comment by tresni** [comment]
Been using this since 2014, still :+1: for me :)

File: plugins/dash/dash.plugin.zsh [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+_dash() {
+  # No sense doing this for anything except the 2nd position and if we haven't
+  # specified which docset to query against
+  if [[ $CURRENT -eq 2 && ! "$words[2]" =~ ":" ]]; then
+    local -a _all_docsets
+    _all_docsets=()
+    # Use defaults to get the array of docsets from preferences
+    # Have to smash it into one big line so that each docset is an element of
+    # our DOCSETS array
+    DOCSETS=("${(@f)$(defaults read com.kapeli.dash docsets | tr -d '\n' | grep -oE '\{.*?\}')}")
```
Comment (wrecklass) [inline plugins/dash/dash.plugin.zsh]: From my command line this is wrong. Should be:

**com.kapeli._dashdoc_**

Typo or did the key change in Dash recently?

**Comment by wrecklass** [comment]
Added comment to the code. Seems the defaults may have changed name:

com.kapeli.dashdoc

**Comment by Kapeli** [comment]
`com.kapeli.dashdoc` is used by Dash 3 and Dash 4. `com.kapeli.dash` is used by Dash 2.

**Comment by mcornella** [comment]
I've merged it (about time) even if there is a mismatch between versions. Open an issue or PR (preferably) about these changes. Not sure how to query for the version of an app in macOS. Also a README for the plugin would be nice.

**Pull request merged**